index.html

```diff
@@ -3410,6 +3410,191 @@ <h3>
         <h2>
 DID URL Dereferencing
         </h2>
+        <p>
+The DID URL dereferencing function dereferences a <a>DID URL</a> into
+a resource with contents depending on the <a>DID URL</a>'s components,
+including the <a>DID method</a>, method-specific identifier, query parameters,
+and fragment. When a <a>DID URL</a> is also a <a>DID</a>, this process
+ performs <a>DID Resolution</a>. The details of how this
+process is accomplished are outside the scope of this specification, but all
+conformant implementations MUST implement two functions which have the
+following abstract forms:
+        </p>
+
+        <p><code>
+dereference ( did-url, did-url-dereferencing-input-metadata ) <br>
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-&gt; ( did-url-dereferencing-metadata, content, content-metadata )
+        </code></p>
+
+        <p>
+The input variables of these functions MUST be as follows:
+        </p>
+
+        <dl>
+            <dt>
+did-url
+            </dt>
+            <dd>
+A conformant <a>DID URL</a> as a single string. This is the <a>DID URL</a> to dereference.
+This input is REQUIRED.
+            </dd>
+            <dt>
+did-url-dereferencing-input-metadata
+            </dt>
+            <dd>
+A <a href="metadata-structure">metadata structure</a> consisting of input
+options to the <code>dereference</code> function in addition to the <code>did-url</code>
+itself.
+Properties defined by this specification are in <a href="#did-url-dereferencing-input-metadata-properties"></a>.
+This input is REQUIRED, but the structure MAY be empty.
+            </dd>
+        </dl>
+
+        <p>
+The output variables of these functions MUST be as follows:
+        </p>
+
+        <dl>
+            <dt>
+did-url-dereferencing-metadata
+            </dt>
+            <dd>
+A <a href="metadata-structure">metadata structure</a> consisting of values
+relating to the results of the <a>DID URL Dereferencing</a> process. This
+structure is REQUIRED and MUST NOT be empty.
+Properties defined by this specification are in 
+<a href="#did-url-dereferencing-metadata-properties"></a>.
+If the dereferencing is not successful, this structure MUST contain an
+<code>error</code> property describing the error.
+            </dd>
+            <dt>
+content
```

Comment:
I propose to rename this to `content-stream`. In my opinion, `dereference()` is more like `resolveStream()` than `resolve()`, since dereferencing returns a concrete representation of a resource. Would you agree or do you see that differently?

If you agree, then you could also add a sentence to the "did-url-dereferencing-metadata" item that says that it MUST contain a `content-type` property (similar to how we say that in the "did-resolution-metadata" item).

Reply:
Do you recommend we use this PR to move the MUST to SHOULD in `resolveStream`, `resolve`, and `dereference`? I will start on this path unless objections arise.

Reply:
I think I'd propose that we make it a SHOULD here for `dereference` (following your suggestion), and then create a separate issue later to discuss whether or not we should do the same for `resolveStream`. I think I remember that back then there was already some discussion about this, but I can't find any references now.

Reply:
Makes sense, and I think it's a good idea. I'll incorporate these changes.